api/libproject/devstart.rb

```diff
@@ -1204,6 +1204,47 @@ def connect_to_cloud_db(cmd_name, *args)
   :fn => ->(*args) { connect_to_cloud_db("connect-to-cloud-db", *args) }
 })
 
+def connect_to_cloud_db_binlog(cmd_name, *args)
+  ensure_docker cmd_name, args
+  common = Common.new
+  op = WbOptionsParser.new(cmd_name, args)
+  gcc = GcloudContextV2.new(op)
+  op.parse.validate
+  gcc.validate
+  env = read_db_vars(gcc)
+  CloudSqlProxyContext.new(gcc.project).run do
+    common.status "\n" + "*" * 80
+    common.status "Listing available journal files: "
+
+    # "root" is required for binlog access.
+    password = env["MYSQL_ROOT_PASSWORD"]
```

Comment:
Is the storage & use of mysql passwords documented anywhere? L1244 doesn't mention it, nor does the runbook. 

Looking around, it's scattered all throughout this file, so I suppose devs are familiar with it... but the runbook should probably have some pointer on where / how to find the root password. Do we have some standard place to store team secrets like this?

Reply:
w.r.t. these tools, it's an implementation detail. The credentials themselves are stored in GCS. Most of these tools read that file dynamically to pull necessary credential information automatically (this happens in `read_db_vars()` above).

There's no expected prerequisite of environment variables etc for running these commands, which is probably what this code looks like.

Reply:
Thanks – my mistake for not reading carefully, for some reason I'd thought 'env' was the shell environment, not a local variable :) Makes total sense.